apps/tangle-cloud/src/data/services/useServiceRequestTx.ts

```diff
@@ -0,0 +1,178 @@
+import createAssetIdEnum from '@tangle-network/tangle-shared-ui/utils/createAssetIdEnum';
+import useAgnosticTx from '@tangle-network/tangle-shared-ui/hooks/useAgnosticTx';
+import { SUCCESS_MESSAGES } from '../../hooks/useTxNotification';
+import { useCallback } from 'react';
+
+import { TxName } from '../../constants';
+import { SubstrateTxFactory } from '@tangle-network/tangle-shared-ui/hooks/useSubstrateTx';
+import SERVICES_PRECOMPILE_ABI from '@tangle-network/tangle-shared-ui/abi/services';
+import { PrecompileAddress } from '@tangle-network/tangle-shared-ui/constants/evmPrecompiles';
+import {
+  EvmAddress,
+  SubstrateAddress,
+} from '@tangle-network/ui-components/types/address';
+import { PrimitiveField } from '@tangle-network/tangle-shared-ui/types/blueprint';
+import { RestakeAssetId } from '@tangle-network/tangle-shared-ui/types';
+import { EvmTxFactory } from '@tangle-network/tangle-shared-ui/hooks/useEvmPrecompileCall';
+import { Hash, zeroAddress } from 'viem';
+import {
+  convertAddressToBytes32,
+  isEvmAddress,
+  toEvmAddress,
+} from '@tangle-network/ui-components';
+import { getApiPromise } from '@tangle-network/tangle-shared-ui/utils/polkadot/api';
+import useNetworkStore from '@tangle-network/tangle-shared-ui/context/useNetworkStore';
+
+export type Context = {
+  blueprintId: bigint;
+  permittedCallers: Array<SubstrateAddress | EvmAddress>;
+  operators: Array<SubstrateAddress>;
+  requestArgs: PrimitiveField[];
+  securityRequirements: Array<{
+    minExposurePercent: number;
+    maxExposurePercent: number;
+  }>;
+  assets: RestakeAssetId[];
+  ttl: bigint;
+  paymentAsset: string;
+  paymentValue: bigint;
+  membershipModel: 'Fixed' | 'Dynamic';
+  minOperator: number;
+  maxOperator: number;
+};
+
+const useServiceRegisterTx = () => {
+  const wsRpcEndpoint = useNetworkStore((store) => store.network.wsRpcEndpoint);
+
+  const substrateTxFactory: SubstrateTxFactory<Context> = useCallback(
+    async (api, activeSubstrateAddress, context) => {
+      const membershipModel =
+        context.membershipModel === 'Fixed'
+          ? {
+              Fixed: {
+                minOperators: context.minOperator,
+              },
+            }
+          : {
+              Dynamic: {
+                minOperators: context.minOperator,
+                maxOperators: context.maxOperator,
+              },
+            };
```

Comment:
Create a reusable function, like `createAssetIdEnum`, to consolidate this logic.